sycl/doc/extensions/experimental/sycl_ext_oneapi_complex.asciidoc

```diff
@@ -0,0 +1,334 @@
+            /// Streams the complex number z in the format "(real,imaginary)" into `sycl::stream` x and return the result.
+            friend const sycl::stream &operator<<(const sycl::stream &x, const complex<value_type> &z);
+
+} // namespace experimental
+} // namespace oneapi
+} // namespace ext
+} // namespace sycl
+```
+
+=== Mathematical operations
+
+This proposal adds to the `sycl::ext::oneapi::experimental` namespace, math
+functions accepting the complex types `complex<sycl::half>`, `complex<float>`,
+`complex<double>` as well as the scalar types `sycl::half`, `float` and `double`
+for the SYCL math functions, `abs`, `acos`, `asin`, `atan`, `acosh`, `asinh`,
+`atanh`, `arg`, `conj`, `cos`, `cosh`, `exp`, `log`, `log10`, `norm`, `polar`,
+`pow`, `proj`, `sin`, `sinh`, `sqrt`, `tan`, and `tanh`.
+
+Each math function should follow the C++ standard for handling NaN's and Inf
+values.
+
+```C++
+namespace sycl {
+namespace ext {
+namespace oneapi {
+namespace experimental {
+
+    /// VALUES:
+    /// Returns the real component of the complex number z.
+    template <class T> constexpr T real(const complex<T> &);
+    /// Returns the real component of the number y, treated as complex numbers with zero imaginary component.
+    template <class T> constexpr T real(T);
+    /// Returns the imaginary component of the complex number z.
+    template <class T> constexpr T imag(const complex<T> &);
+    /// Returns the imaginary component of the number y, treated as complex numbers with zero imaginary component.
+    template <class T> constexpr T imag(T);
+
+    /// Compute the magnitude of complex number x.
+    template <class T> T abs(const complex<T> &);
+    /// Compute phase angle in radians of complex number x.
+    template <class T> T arg(const complex<T> &);
+    /// Compute phase angle in radians of complex number x, treated as complex number with positive zero imaginary component.
+    template <class T> T arg(T);
+    /// Compute the squared magnitude of complex number x.
+    template <class T> T norm(const complex<T> &);
+    /// Compute the squared magnitude of number x, treated as complex number with positive zero imaginary component.
+    template <class T> T norm(T);
+    /// Compute the conjugate of complex number x.
+    template <class T> complex<T> conj(const complex<T> &);
+    /// Compute the conjugate of number y, treated as complex number with positive zero imaginary component.
+    template <class T> complex<T> conj(T);
+    /// Compute the projection of complex number x.
+    template <class T> complex<T> proj(const complex<T> &);
+    /// Compute the projection of number y, treated as complex number with positive zero imaginary component.
+    template <class T> complex<T> proj(T);
+    /// Construct a complex number from polar coordinates with mangitude rho and angle theta.
+    template <class T> complex<T> polar(const T &, const T &);
```

Comment:
Note that the standard C++ version of `polar` has a default value for the second parameter (theta).  Did you intend to do that here also?

Reply:
Good catch; the implementation does have a default value for `theta`. I'll add it to the extension :+1: